include/ofxColour.h

```diff
@@ -0,0 +1,259 @@
+#ifndef _ofxColour_h_
+#define _ofxColour_h_
+
+// Copyright OpenFX and contributors to the OpenFX project.
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @file ofxColour.h
+Contains the API for colourspace data exchange.
+*/
+
+/** @brief What style of colour management does the host or plug-in offer?
+
+   - Type - string X 1
+   - Property Set - host descriptor (read only), plugin descriptor (read/write), image effect instance (read only)
+   - Default - kOfxImageEffectPropColourManagementStyleNone
+   - Valid Values - This must be one of
+     - ::kOfxImageEffectPropColourManagementNone - no colour management
+     - ::kOfxImageEffectPropColourManagementBasic - only basic colourspaces from the config header may be used
+     - ::kOfxImageEffectPropColourManagementCore - only core colourspaces from the config header may be used
+     - ::kOfxImageEffectPropColourManagementFull - any colourspace from the config header may be used
+     - ::kOfxImageEffectPropColourManagementOCIO - any OCIO config may be used (implies use of the OCIO library)
+
+Hosts should set this property if they will provide colourspace information 
+to plug-ins.
+Plug-ins should set this property if they can use host-provided colourspace 
+information.
+Collectively, the full, core and basic styles are referred to as native
+colour management. OCIO is used as the reference for the colour management
+API, but is not required to implement the native styles.
+
+The colourspace strings used in the native styles are from ofx-native-1.5.ocio,
+which is based on the OCIO ACES Studio built-in config,
+studio-config-v2.1.0_aces-v1.3_ocio-v2.3, and stored for OFX purposes in
+openfx-native-v1.5_aces-v1.3_ocio-v2.3.h (referred to as the config header).
+Additionally, there is a scheme for cross-referencing between clips, and a set
+of "basic colourspaces", which are designed to be generic names for a family of
+colourspaces. When a basic colourspace is used, this means the host is free to
+choose any colourspace with the same encoding and reference space (scene vs
+display).
+
+The assumption is that OCIO > Full > Core > Basic, so the highest style
+supported by both host and plug-in should usually be chosen by the host.
+The chosen style must be set using this property on an image effect instance.
+*/
```

Comment:
It might be worth clarifying that, even if both sides are in OCIO mode, the Basic color space indicators may still be used to indicate that the other side has the flexibility to provide any color space in the given family.

Reply:
I hadn't thought about that since removing the basic colourspace roles from the OCIO config. I've now permitted the use of basic colourspaces in `kOfxImageClipPropPreferredColourspaces` but not in `kOfxImageClipPropColourspace`, on the basis that in OCIO mode it should always be possible to label the clip with the actual colourspace that was used.

If the host has actually added roles or aliases into its config for the basic colourspaces, they would then be legal to pass in `kOfxImageClipPropColourspace`. I've clarified the docs to cover this.

Reply:
@doug-walker  - I have a question about basic scene log just to be sure - is this equivalent to ACEScc curve excluding a particular chromacity (also for negative values when in float) ... straight log 2 where black and white points are nornalized to 0-1?

Reply:
@revisionfx , my interpretation of "ofx_scene_log" is that it is a request to exchange any generic camera-log color space, which would include ARRI LogC (any version), Sony S-Log*, RED Log3G10, ACEScct, and etc. I think ACEScc would certainly fall in that group, but it certainly would not have to use that particular curve.

Many, though not all, plug-ins that work on generic video pixels will work on "ofx_scene_log" too, so I would guess that many plug-ins would include this in their list of acceptable inputs and this would give the host some useful flexibility in what it may provide.